api/views.py

```diff
@@ -509,6 +509,9 @@ def set_subject(task):
         else:
             task.plagiarism_key = plagiarism_key[:255]
 
+        if data.get("type"):
```

Comment:
If the type will be invalid, this will end in 500 server error. Let's check if the type is valid based on the `TextChoices` values, and if not, return a proper 400 bad client request error.